internal/operands/common-templates/reconcile.go

```diff
@@ -90,6 +111,16 @@ func (c *commonTemplates) Reconcile(request *common.Request) ([]common.Reconcile
 	return append(reconcileTemplatesResults, oldTemplatesResults...), nil
 }
 
+func getClusterArchitectures(sspObj *ssp.SSP) []string {
+	if sspObj.Spec.EnableMultipleArchitectures != nil && *sspObj.Spec.EnableMultipleArchitectures {
+		return sspObj.Spec.Cluster.WorkloadArchitectures
+	}
+	if sspObj.Spec.Cluster != nil {
```

Comment:
`!= nil` doesn't mean it's not empty.

Reply:
Check for `len(sspObj.Spec.Cluster) > 0`

Reply:
`.Spec.Cluster` is not a slice, its type is `*Cluster`.

Reply:
I see. Those are checked in webhook. But I can add the check here.

Reply:
Sorry, for confusion.
I see now that if the arch slice is empty, then the reconcile will fail, because of this block of code:
```
	clusterArchs := getClusterArchitectures(request.Instance)
	if len(clusterArchs) == 0 {
		return []common.ReconcileResult{}, fmt.Errorf("no cluster architectures defined")
	}
```

Is it enough?

Reply:
I will fix the nil ptr.